### PR TITLE
[ng] Move testing prose from toh-pt0 to Testing pag

### LINKED ISSUE #506: Move testing text out of toh-0
See #478  for details.

### PR DESCRIPTION
**Pull request by chalin**
Fixes #506

Staged at: 
- https://angulardart-org-dev.firebaseapp.com/angular/tutorial/toh-pt0#the-starter-apps-code
- https://angulardart-org-dev.firebaseapp.com/angular/guide/testing

### CODE REVIEW

File: src/angular/guide/testing.jade [hunk 1]
```diff
@@ -16,16 +16,13 @@ include ../_util-fns
 .l-main-section
 :marked
   Testing is an essential part of software development.
-  This page offers a basic introduction to
-  **[component / subsystem testing][testing]**
-  of Angular apps using the **[angular_test][]** package.
-  You'll learn how to use this package
-  by writing tests for the [tutorial](../tutorial)'s _**Tour of Heroes**_ app.
+  This page shows how to use the **[angular_test][]** package to perform
+  **[component/subsystem testing][testing]**.
```
Comment (chalin) [inline src/angular/guide/testing.jade]: Thanks, I like the simplifications you've made, though now we lose the scope/context-setting phrase that the reader will be
> writing tests for the [tutorial](../tutorial)'s _**Tour of Heroes**_ app.

Or maybe you don't think that that is essential.

File: src/angular/tutorial/toh-pt0.jade [hunk 2]
```diff
@@ -110,6 +115,14 @@ table(width="100%")
         Defines `<my-app>`, the **root** component of what will become a tree of nested components
         as the application evolves.
   tr
+    td <code>test/app_test.dart</code>
+    td
+      :marked
+        Defines `AppComponent` tests.
+        While testing isn't covered in the tutorial,
```
Comment (kwalrath) [inline src/angular/tutorial/toh-pt0.jade]: the tutorial -> this tutorial

File: src/angular/guide/testing.jade [hunk 3]
```diff
@@ -13,9 +13,116 @@ nextpage:
...
+
+- var _example = 'toh-1';
+:marked
+  ## Test the Hero Editor
+
+  In this section you'll write tests for the
+  [tutorial](../tutorial)'s
+  [Hero Editor](../tutorial/toh-pt1).
+
+  TBC
```
Comment (kwalrath) [inline src/angular/guide/testing.jade]: Let's spell it out.

_To be continued..._

File: src/angular/guide/testing.jade (hunk 1 above)
Comment (kwalrath) [inline src/angular/guide/testing.jade]: It didn't seem essential to me, and it's repeated below.

File: src/angular/guide/testing.jade [hunk 4]
```diff
@@ -13,9 +13,116 @@ nextpage:
...
+
+  [test fixture]: https://github.com/junit-team/junit4/wiki/test-fixtures
+
+  The file contains a test named "Default greeting" that checks whether
+  the app component template generates the text "Hello Angular".
+
++makeExcerpt('test/app_test.dart', 'default-test', '')
+
+:marked
+  As you'd guess from the test file imports, the app's `pubspec.yaml` has
```
Comment (chalin) [inline src/angular/guide/testing.jade]: Good. I used that with just a tweak (app depends --> test file depends).

File: src/angular/guide/testing.jade [hunk 5]
```diff
@@ -13,9 +13,116 @@ nextpage:
...
+  the [test][] and [angular_test][] packages listed under `dev_dependencies`.
+
+  [test]: https://pub.dartlang.org/packages/test
+  [angular_test]: https://pub.dartlang.org/packages/angular_test
+
++makeExcerpt('pubspec.yaml (dev_dependencies)')
+
+:marked
+  More importantly, the `pubspec.yaml` file also includes
```
Comment (chalin) [inline src/angular/guide/testing.jade]: :) fixed.

File: src/angular/guide/testing.jade [hunk 6]
```diff
@@ -13,9 +13,116 @@ nextpage:
...
+  00:03 +0: test/app_test.dart: Default greeting
+  00:03 +1: test/app_test.dart: Default greeting
+  00:03 +1: test/app_test.dart: Greet world
+  00:03 +2: test/app_test.dart: Greet world
+  00:03 +2: test/app_test.dart: Greet world HTML
+  00:03 +3: test/app_test.dart: Greet world HTML
+  00:03 +3: All tests passed!
+
+:marked
+  Take a look at the remaining tests:
```
Comment (chalin) [inline src/angular/guide/testing.jade]: Because the first code excerpt in the page only shows the first test.

File: src/angular/guide/testing.jade [hunk 7]
```diff
@@ -13,9 +13,116 @@ nextpage:
...
+
++makeExcerpt('test/app_test.dart', 'more-tests', '')
+
+:marked
+  The first of these tests interacts with the app component fixture by setting
```
Comment (chalin) [inline src/angular/guide/testing.jade]: There is a link to a definition of fixture at the start of the page. Is that enough?

File: src/angular/guide/testing.jade [hunk 8]
```diff
@@ -13,9 +13,116 @@ nextpage:
...
+  the component's `name` property to "World", and expecting the infamous
```
Comment (chalin) [inline src/angular/guide/testing.jade]: Right. Fixed.

File: src/angular/guide/testing.jade (hunk 6 above)
Comment (chalin) [inline src/angular/guide/testing.jade]: I can change the text to: "Take a look at the other tests in the test file:".

Comment (kwalrath) [inline src/angular/guide/testing.jade]: Ah, that was a long time ago. Out of sight, out of mind. :)

How about:

You've already seen the code for the "Default greeting" test. Here's the code for the "Greet world" and "Greet world HTML" tests:

File: src/angular/guide/testing.jade [hunk 9]
```diff
@@ -13,9 +13,116 @@ nextpage:
...
+  greeting as a result.
+
+  The second test uses the fixture's `rootElement` to access the native DOM
```
Comment (chalin) [inline src/angular/guide/testing.jade]: Fixed (doing the former).

File: src/angular/guide/testing.jade (hunk 3 above)
Comment (chalin) [inline src/angular/guide/testing.jade]: Fixed

File: src/angular/tutorial/toh-pt0.jade (hunk 2 above)
Comment (chalin) [inline src/angular/tutorial/toh-pt0.jade]: Fixed

File: src/angular/guide/testing.jade (hunk 7 above)
Comment (kwalrath) [inline src/angular/guide/testing.jade]: I do remember seeing "test fixture", but "app component fixture" threw me. Maybe either "app component's test fixture" or just "test fixture". Or set up the term "app component fixture" earlier.

Also, "first of these tests" is a little hard to parse. Maybe either:
first test
_or_
"Greet world" test

(btw, I keep accidentally typing "text" instead of "test". Take any "text" from me with a grain of salt.)

File: src/angular/guide/testing.jade (hunk 6 above)
Comment (chalin) [inline src/angular/guide/testing.jade]: Done.

File: src/angular/guide/testing.jade (hunk 7 above)
Comment (chalin) [inline src/angular/guide/testing.jade]: I agree with your additional comment. I'll adjust further.
